### PR TITLE
how to create SSL key

### PR DESCRIPTION
**Pull request by NickyDaB**
working on : https://github.com/NickyDaB/CAST/issues/31

document how to connect to UFM restfulAPIs via SSL connection

### CODE REVIEW

File: docs/source/csm-integration/Mellanox_and_Infiniband.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+=======================
+
+Network Inventory Collection
+----------------------------
+
+The network inventoy (such as: switches, switch modules, and cables) can be collected and stored into the `CSM Database`_. 
+
+.. _CSM Database: https://cast.readthedocs.io/en/latest/csmdb/index.html
+
+Inventory collection has been modularly developed. We seperated the external inventory data collection from the internal CSM Database insertion. Once data has been collected, you can then insert that collected data into the CSM Database by using a CSM API. We do this for ease of future updates, should an external component change the way it collects and presents its data.
```
Comment (mew2057) [inline docs/source/csm-integration/Mellanox_and_Infiniband.rst]: seperated => separated

File: docs/source/csm-integration/Mellanox_and_Infiniband.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+For more information on generic CSM Inventory collection, read this document: ``LINK_TO_CSM_INVENTORY_SECTION``.
+
+Because of this development choice, CSM can easily be adapted to work with multiple external programs and services. As of CSM 1.0, CSM supports Mellanox and IB inventory collection. For ease of use, CSM provides a tool: ``standalone_ib_and_switch_collection`` which will collect inventory information from Mellanox and insert that data into the CSM Database. 
+
+Credentials
+-----------
+
+To communicate with the Mellanox UFM restfulAPIs, a user must have proper credentials. CSM will attempt to connect to UFM via an SSL key. The location of your SSL key can be configured via the ``csm_master.cfg`` file. Using the ``ufm_ssl_file_path`` and ``ufm_ssl_file_name`` fields. Default values have been reproduced below for reference.
+
+.. code-block:: json
```
Comment (mew2057) [inline docs/source/csm-integration/Mellanox_and_Infiniband.rst]: Does this still read as invalid? If so Change to JavaScript or remove the comma at the end.

File: docs/source/csm-integration/Mellanox_and_Infiniband.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    "ufm" :
+    {
+        "rest_address"  : "__UFM_REST_ADDRESS__",
+        "rest_port"     : 80,
+        "ufm_ssl_file_path" : "/etc/ibm/csm",
+        "ufm_ssl_file_name" : "csm_ufm_ssl_key.txt"
+    }, 
+
+An SSL key must be generated and placed in this file or CSM will not be able to communicate with UFM restfulAPIs.
```
Comment (mew2057) [inline docs/source/csm-integration/Mellanox_and_Infiniband.rst]: restful APIs

File: docs/source/csm-integration/Mellanox_and_Infiniband.rst (hunk 1 above)
Comment (NickyDaB) [reply docs/source/csm-integration/Mellanox_and_Infiniband.rst]: good catch

File: docs/source/csm-integration/Mellanox_and_Infiniband.rst (hunk 3 above)
Comment (NickyDaB) [reply docs/source/csm-integration/Mellanox_and_Infiniband.rst]: another good catch

File: docs/source/csm-integration/Mellanox_and_Infiniband.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        "rest_address"  : "__UFM_REST_ADDRESS__",
+        "rest_port"     : 80,
+        "ufm_ssl_file_path" : "/etc/ibm/csm",
+        "ufm_ssl_file_name" : "csm_ufm_ssl_key.txt"
+    }, 
+
+An SSL key must be generated and placed in this file or CSM will not be able to communicate with UFM restfulAPIs. 
+
+SSL key generation can be done via the ``openssl`` command found in UNIX. Creating a key for the default username and password is shown below:
```
Comment (pdlun92) [inline docs/source/csm-integration/Mellanox_and_Infiniband.rst]: Follow up on offline discussion, just wanted to document suggestion that this sentence explicitly state `admin` is the username and `123456` is the password for the subsequent example

File: docs/source/csm-integration/Mellanox_and_Infiniband.rst (hunk 2 above)
Comment (NickyDaB) [reply docs/source/csm-integration/Mellanox_and_Infiniband.rst]: it did. I went back and surrounded the json with opening and closing brackets.

**Comment by NickyDaB** [comment]
updated with @pdlun92 and @mew2057  suggestions

**Pull request merged**